webroot/src/network/licenseApi/data.api.ts

```diff
@@ -100,38 +108,62 @@ export class LicenseDataApi implements DataApiInterface {
      * @return {object}                             The URI query param object.
      */
     public prepRequestPostParams(params: RequestParamsInterfaceLocal = {}): RequestParamsInterfaceRemote {
+        const {
+            compact,
+            jurisdiction,
+            licenseeId,
+            licenseeFirstName,
+            licenseeLastName,
+            licenseeSsn,
+            pageSize,
+            lastKey,
+            sortBy,
+            sortDirection,
+        } = params;
+        const hasSearchTerms = Boolean(licenseeId || licenseeFirstName || licenseeLastName || licenseeSsn);
         const requestParams: RequestParamsInterfaceRemote = { query: {}};
 
         if (params.compact) {
-            requestParams.query.compact = params.compact;
+            requestParams.query.compact = compact;
         }
 
-        if (params.jurisdiction) {
-            requestParams.query.jurisdiction = params.jurisdiction;
+        if (jurisdiction) {
+            requestParams.query.jurisdiction = jurisdiction;
         }
 
-        if (params.licenseeId) {
-            requestParams.query.providerId = params.licenseeId;
+        if (hasSearchTerms) {
+            if (licenseeId) {
+                requestParams.query.providerId = licenseeId;
```

Comment:
Any reason to rename (or not to rename) `licensee` to `provider` across the codebase?

Reply:
In team calls and the UI copy / labels we seem to use "licensee" & "license". For the frontend part, it feels the least strain to use those terms.

I'm generally ok with the UI & Data API each doing what they need to do, as we've got the mesh point distilled down to a single serializer function in each frontend model. Plus the API docs as backup. Happy to discuss further if needed.

Reply:
That sounds good, thanks for the explanation!